From ee79132e06852213646775fdfd591616e666bb44 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Tue, 23 Feb 2016 17:55:35 -0500
Subject: [PATCH 053/134] trace-cmd profile: Handle NULL case in
 handle_fgraph_exit_event()

If there's no memory available, handle_end_event() will return a NULL task,
which needs to be checked before it is dereferenced.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-profile.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/trace-profile.c b/trace-profile.c
index 7911ca1baa3b..0b0444db9436 100644
--- a/trace-profile.c
+++ b/trace-profile.c
@@ -1179,6 +1179,8 @@ static int handle_fgraph_exit_event(struct handle_data *h,
 	struct task_data *task;
 
 	task = handle_end_event(h, event_data, record, pid);
+	if (!task)
+		return -1;
 	/* Do not match stacks with function graph exit events */
 	task->last_event = NULL;
 
-- 
2.5.5

